gensim/models/ldaseqmodel.py

```diff
@@ -0,0 +1,1121 @@
+        self.lhood = lhood
+
+        if max_nterms is not None and num_topics is not None:
+            self.phi = numpy.resize(numpy.zeros(max_nterms * num_topics), (max_nterms, num_topics))
+            self.log_phi = numpy.resize(numpy.zeros(max_nterms * num_topics), (max_nterms, num_topics))
+
+        # the following are class variables which are to be integrated during Document Influence Model
+
+        self.doc_weight = None
+        self.renormalized_doc_weight = None
+
+def update_zeta(sslm):
+
+    num_terms = sslm.num_terms
+    num_sequences = sslm.num_sequences
+    # making zero and updating
+    sslm.zeta.fill(0)
+    for i in range(0, num_terms):
+        for j in range(0, num_sequences):
+            m = sslm.mean[i][j + 1]
+            v = sslm.variance[i][j + 1]
+            val = numpy.exp(m + v/2)
+            sslm.zeta[j] = sslm.zeta[j] + val  
+    return
+
+def compute_post_variance(word, sslm, chain_variance):
+
+    T = sslm.num_sequences
+    variance = sslm.variance[word] # pick wordth row
+    fwd_variance = sslm.fwd_variance[word] # pick wordth row
+
+    # forward pass. Set initial variance very high
+    fwd_variance[0] = chain_variance * 1000
+    
+    for t in range(1, T + 1):
+        if sslm.obs_variance:
+            w = sslm.obs_variance / (fwd_variance[t - 1] + chain_variance + sslm.obs_variance)
+        else:
+            w = 0
+        fwd_variance[t] = w * (fwd_variance[t-1] + chain_variance)
+
+    # backward pass 
+    variance[T] = fwd_variance[T]
+    for t in range(T - 1, -1, -1):
+        if fwd_variance[t] > 0.0:
+            w = numpy.power((fwd_variance[t] / (fwd_variance[t] + chain_variance)), 2)
+        else:
+            w = 0
+        variance[t] = (w * (variance[t + 1] - chain_variance)) + ((1 - w) * fwd_variance[t])
+
+    sslm.variance[word] = variance
+    sslm.fwd_variance[word] = fwd_variance
+    return
+    
+
+def compute_post_mean(word, sslm, chain_variance):
+
+    T = sslm.num_sequences
+    obs = sslm.obs[word] # wordth row
+    mean = sslm.mean[word]
+    fwd_mean = sslm.fwd_mean[word]
+    fwd_variance = sslm.fwd_variance[word]
+
+    # forward 
+    fwd_mean[0] = 0
+    for t in range(1, T + 1):
+        w = sslm.obs_variance / (fwd_variance[t - 1] + chain_variance + sslm.obs_variance)
+        fwd_mean[t] = w * fwd_mean[t - 1] + (1 - w) * obs[t - 1]
+        if fwd_mean[t] is None:
+            # error message
+            pass
+
+    # backward pass
+    mean[T] = fwd_mean[T]
+    for t in range(T - 1, -1, -1):
+        if chain_variance == 0.0:
+            w = 0.0
+        else:
+            w = chain_variance / (fwd_variance[t] + chain_variance)
+        mean[t] = w * fwd_mean[t] + (1 - w) * mean[t + 1]
+        if mean[t] is None:
+            # error message
+            pass
+ 
+    return
+
+def compute_expected_log_prob(sslm):
+
+    W = sslm.num_terms
+    T = sslm.num_sequences
+    for t in range(0, T):
+        for w in range(0, W):
```

Comment:
In come cases such as in compute_post_mean where I iterate to a num_sequence + 1th slice for mean and fwd_mean, I would need it. The places I don't I could use enumerate. Is that ok?